bears/python/MypyBear.py

```diff
@@ -0,0 +1,157 @@
+    AUTHORS = {'Petr Viktorin'}
+    REQUIREMENTS = {PipRequirement('mypy', '0.*')}
+    AUTHORS_EMAILS = {'encukou@gmail.com'}
+    LICENSE = 'AGPL-3.0'
+
+    # This detects typing errors, which is pretty unique -- it doesn't
+    # make sense to add a category for it.
+    CAN_DETECT = set()
+
+    @add_param_docs(FLAG_MAP)
+    def create_arguments(self, filename, file, config_file,
+                         language: str="Python 3",
+                         python_version: str=None,
+                         allow_untyped_functions: bool=True,
+                         allow_untyped_calls: bool=True,
+                         check_untyped_function_bodies: bool=False,
+                         strict_optional: bool=False,
+                         ):
+        """
+        :param language:
+            Set to "Python" or "Python 3" to check Python 3.x source.
+            Use "Python 2" for Python 2.x.
+        :param python_version:
+            Set the specific Python version, e.g. "3.5".
+        """
+        args = ['-m', 'mypy']
+        if language.lower() == 'python 2':
+            args.append('--py2')
+        elif language.lower() not in ('python 3', 'python'):
+            # Ideally, this would fail the check, but I don't know of a good
+            # way to fail from create_arguments.
+            # See https://github.com/coala-analyzer/coala/issues/2573
+            self.err(
+                'Language needs to be "Python", "Python 2" or "Python 3". '
+                'Assuming Python 3.')
+        if python_version:
+            args.extend(['--python-version', python_version])
+        loc = locals()
+        args.extend(flag.arg for name, flag in FLAG_MAP.items()
+                    if flag.want_flag(loc[name]))
+        args.append(filename)
+        return args
+
+    def process_output(self, output, filename, file):
+        # Mypy generates messages in the format:
+        #    blabla.py: note: In function "f":
+        #    blabla.py:2: error: Unsupported operand types for ...
+        # The "note" messages are only adding info coala should already know,
+        # so discard those.
+        # Also discard results for files other than the one we're currently
+        # checking.
+
+        for match in re.finditer(OUTPUT_REGEX, output):
+            message_filename = match.group('filename').strip()
+            message_filename = os.path.abspath(message_filename)
+            line = match.group('line')
+            if line is not None:
+                line = int(line)
+            if (match.group('severity') != 'note' and
+                    message_filename == os.path.abspath(filename)):
+                result = Result.from_values(
+                    origin=self,
+                    message=match.group('message'),
+                    file=message_filename,
+                    severity=RESULT_SEVERITY.MAJOR,
+                    line=line,
+                    )
+                yield result
```

Comment:
quilckly looking over it, it seems you perform a regex pattern matching. Can't you use `output-format='regex'` inside `@linter` and provide suited regexes?